aexpect/client.py

```diff
@@ -277,6 +277,24 @@ def _close_reader_fds(self):
             except OSError:
                 pass
 
+    def _close_aexpect_helper(self):
+        """
+        Makes sure the subprocess holding the aexpect_helper is terminated
+        and that the file descriptors related to it are closed.
+        """
+        if self._aexpect_helper:
+            if self._aexpect_helper.stdout:
+                self._aexpect_helper.stdout.close()
+            try:
+                if self._aexpect_helper.stdin:
+                    self._aexpect_helper.stdin.close()
+            finally:
+                try:
+                    self._aexpect_helper.wait(10)
+                except subprocess.TimeoutExpired:
+                    self._aexpect_helper.kill()
+                    self._aexpect_helper.wait(10)
```

Comment:
To copy the exact way that `subprocess.call` handled this, I'd suggest calling `wait`/`kill` at first:

```suggestion
            try:
                try:
                    self._aexpect_helper.wait(10)
                except subprocess.TimeoutExpired:
                    self._aexpect_helper.kill()
            finally:
                if self._aexpect_helper.stdout:
                    self._aexpect_helper.stdout.close()
                try:
                    if self._aexpect_helper.stdin:
                        self._aexpect_helper.stdin.close()
                finally:
                    self._aexpect_helper.wait(10)
```


https://github.com/python/cpython/blob/be578e0c063dad1dbb273f86d5bc77e4e6f14583/Lib/subprocess.py#L345-L351
https://github.com/python/cpython/blob/be578e0c063dad1dbb273f86d5bc77e4e6f14583/Lib/subprocess.py#L1033-L1059

Reply:
I'll redo the commit right now, thank you